Sources/MintKit/PackageReference.swift

```diff
@@ -3,9 +3,54 @@ import PathKit
 
 public class PackageReference {
     public var repo: String
-    public var version: String
+    public var version: Revision
 
-    public init(repo: String, version: String = "") {
+    public enum Revision: Equatable {
+        case tag(name: String)
+        case branch(name: String)
+        case commit(hash: String)
+        case unspecified
+
+        var string: String {
+            switch self {
+            case .tag(name: let name), .branch(name: let name), .commit(hash: let name):
+                return name
+            case .unspecified:
+                return ""
+            }
+        }
+
+        init(versionString: String) {
+            let parts = versionString.split(separator: ":")
+
+            if parts.count == 1 {
+                self = .branch(name: versionString)
```

Comment:
Can we add a new case `unknown`. If the string has 2 dots in it though let's assume it's `.tag`